File: 4-pointers/example_1.go
package main

import "fmt"

func main() {
	x := 1
	y := x // copy

	y = 100

	fmt.Println("x:", x) // x == 1 (unchanged)
	fmt.Println("y:", y) // y == 100

	p := &x
	*p = 100

	fmt.Println("x after *p = 100:", x) // x == 100 (changed)
}

File: 4-pointers/Example1.java
public class Example1 {
    public static void main(String[] args) {
        int x = 1;
        int y = x; // copy

        y = 100;

        System.out.println("x: " + x); // x == 1 (unchanged)
        System.out.println("y: " + y); // y == 100
    }
}
